### ISSUE TITLE
furl.url does not work properly with unicode in the url

### ISSUE THREAD

**Issue by lisimia** [gruns/furl#72]
````
In [16]: furl(u'/Assassin™-Harrows-80%-Tungsten-Soft-Tip-Darts-568-HDT1007.html')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-16-72ee7b05fd36> in <module>()
----> 1 furl(u'/Assassin™-Harrows-80%-Tungsten-Soft-Tip-Darts-568-HDT1007.html')

/usr/local/lib/python2.7/site-packages/furl/furl.py in __init__(self, url, strict)
    860         self.strict = strict
    861
--> 862         self.load(url)  # Raises ValueError on invalid url.
    863
    864     def load(self, url):

/usr/local/lib/python2.7/site-packages/furl/furl.py in load(self, url)
    873
    874         if not isinstance(url, str):
--> 875             url = str(url)
    876
    877         # urlsplit() raises a ValueError on malformed IPv6 addresses in

UnicodeEncodeError: 'ascii' codec can't encode character u'\u2122' in position 9: ordinal not in range(128)
+
````

````
In [1]: furl('http://www.wayfair.com/Assassin%25E2%2584%25A2-Harrows-80%2525-Tungsten-Soft-Tip-Darts-568-HDT1007.html')
Out[1]: ---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
/Library/Python/2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    693                 type_pprinters=self.type_printers,
    694                 deferred_pprinters=self.deferred_printers)
--> 695             printer.pretty(obj)
    696             printer.flush()
    697             return stream.getvalue()

/Library/Python/2.7/site-packages/IPython/lib/pretty.pyc in pretty(self, obj)
    399                             if callable(meth):
    400                                 return meth(obj, self, cycle)
--> 401             return _default_pprint(obj, self, cycle)
    402         finally:
    403             self.end_group()

/Library/Python/2.7/site-packages/IPython/lib/pretty.pyc in _default_pprint(obj, p, cycle)
    519     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    520         # A user-provided repr. Find newlines and replace them with p.break_()
--> 521         _repr_pprint(obj, p, cycle)
    522         return
    523     p.begin_group(1, '<')

/Library/Python/2.7/site-packages/IPython/lib/pretty.pyc in _repr_pprint(obj, p, cycle)
    701     """A pprint that just redirects to the normal repr function."""
    702     # Find newlines and replace them with p.break_()
--> 703     output = repr(obj)
    704     for idx,output_line in enumerate(output.splitlines()):
    705         if idx:

/usr/local/lib/python2.7/site-packages/furl/furl.py in __repr__(self)
   1266
   1267     def __repr__(self):
-> 1268         return "%s('%s')" % (self.__class__.__name__, str(self))
   1269
   1270

/usr/local/lib/python2.7/site-packages/furl/compat.py in __str__(self)
     21     else:  # Python 2
     22         def __str__(self):
---> 23             return self.__unicode__().encode('utf8')

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 31: ordinal not in range(128)
````
However the follow works correctly
````
In [20]: f = furl('/Assassin%25E2%2584%25A2-Harrows-80%2525-Tungsten-Soft-Tip-Darts-568-HDT1007.html')
+
In [21]: f.path
Out[21]: Path('/Assassin™-Harrows-80%-Tungsten-Soft-Tip-Darts-568-HDT1007.html')
````

**Comment by gruns** [comment]
Wonderful bug find. furl should gracefully accept Unicode input -- both in hosts
(see Issue #73), paths (this Issue), and elsewhere.

I'll fix this shortly. Or, pull requests welcome.

**Comment by gruns** [comment]
This bug is fixed in furl v0.5.

```python
>>> import furl
>>> furl.__version__
'0.5'
>>> f = furl.furl(u'/Assassin™-Harrows-80%-Tungsten-Soft-Tip-Darts-568-HDT1007.html')
>>> f.url
'/Assassin%E2%84%A2-Harrows-80%25-Tungsten-Soft-Tip-Darts-568-HDT1007.html'
```

Update with

```
pip install --upgrade furl
```

Thank you for bringing this issue to my attention Lawrence . Don't hesitate to
let me know if there's anything else I can do for you.
